From: KY Support <support@ky.com>
Subject: [PATCH] riscv: cpufeature: guard riscv_isainfo struct definition

Guard struct riscv_isainfo definition and define the guard macro to
prevent redefinition error when hwcap.h is included later.

Signed-off-by: KY Support <support@ky.com>
---
 arch/riscv/include/asm/cpufeature.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/arch/riscv/include/asm/cpufeature.h
+++ b/arch/riscv/include/asm/cpufeature.h
@@ -21,9 +21,12 @@ struct riscv_cpuinfo {
 	unsigned long mimpid;
 };

+#ifndef __RISCV_ISAINFO_DEFINED
+#define __RISCV_ISAINFO_DEFINED
 struct riscv_isainfo {
 	DECLARE_BITMAP(isa, RISCV_ISA_EXT_MAX);
 };
+#endif

 DECLARE_PER_CPU(struct riscv_cpuinfo, riscv_cpuinfo);

